avogadro/io/cmlformat.cpp

```diff
@@ -431,8 +431,9 @@ bool CmlFormat::read(std::istream& file, Core::Molecule& mol)
 
 bool CmlFormat::write(std::ostream& out, const Core::Molecule& mol)
 {
+  char* old_locale = std::setlocale(LC_NUMERIC, NULL);
+  std::setlocale(LC_NUMERIC, "C");
```

Comment:
This is global, and Avogadro is multi-threaded. I was hoping that we could use imbue to set the locale on the stream for example, see https://stdcxx.apache.org/doc/stdlibug/24-3.html and https://pugixml.org/docs/manual.html for some background. It may turn out that C functions are used in places and there is no choice, it would be good to explore.

Reply:
See my comments below - this seems to be the only solution with pugixml.